iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergStorageHandlerWithMultipleCatalogs.java

```diff
@@ -128,13 +130,55 @@ public void testJoinTablesFromDifferentCatalogs() throws IOException {
             HiveIcebergTestUtils.valueForRow(HiveIcebergStorageHandlerTestUtils.CUSTOMER_SCHEMA, rows), 0);
   }
 
+  @Test
+  public void testCTASFromOtherCatalog() throws IOException {
+    testTables2.createTable(shell, "source", HiveIcebergStorageHandlerTestUtils.CUSTOMER_SCHEMA,
+        fileFormat2, HiveIcebergStorageHandlerTestUtils.CUSTOMER_RECORDS);
+
+    shell.executeStatement(String.format(
+        "CREATE TABLE target STORED BY '%s' TBLPROPERTIES ('%s'='%s') AS SELECT * FROM source",
+        HiveIcebergStorageHandler.class.getName(),
+        InputFormatConfig.CATALOG_NAME, HIVECATALOGNAME));
+
+    List<Object[]> objects = shell.executeStatement("SELECT * FROM target");
+    Assert.assertEquals(3, objects.size());
+
+    Table target = testTables1.loadTable(TableIdentifier.of("default", "target"));
+    HiveIcebergTestUtils.validateData(target, HiveIcebergStorageHandlerTestUtils.CUSTOMER_RECORDS, 0);
+  }
+
+  @Test
+  public void testCTASFromOtherCatalogFailureRollback() throws IOException {
+    // force an execution error by passing in a committer class that Tez won't be able to load
+    shell.setHiveSessionValue("hive.tez.mapreduce.output.committer.class", "org.apache.NotExistingClass");
+
+    TableIdentifier target = TableIdentifier.of("default", "target");
+    testTables2.createTable(shell, "source", HiveIcebergStorageHandlerTestUtils.CUSTOMER_SCHEMA,
+        fileFormat2, HiveIcebergStorageHandlerTestUtils.CUSTOMER_RECORDS);
+
+    try {
+      shell.executeStatement(String.format(
+          "CREATE TABLE target STORED BY '%s' TBLPROPERTIES ('%s'='%s') AS SELECT * FROM source",
+          HiveIcebergStorageHandler.class.getName(),
+          InputFormatConfig.CATALOG_NAME, HIVECATALOGNAME));
+    } catch (Exception e) {
+      // expected error
+    }
+
+    // CTAS table should have been dropped by the lifecycle hook
+    Assert.assertThrows(NoSuchTableException.class, () -> testTables1.loadTable(target));
+  }
+
   private void createAndAddRecords(TestTables testTables, FileFormat fileFormat, TableIdentifier identifier,
```

Comment:
We might want to check if we can move this method to TestTables. We have several variations of `createTable` already there.

Reply:
Agreed. We've had a conversation about this testTables cleanup with @lcspinter too. I'll file a ticket for that work.